cpp/arcticdb/async/tasks.hpp

```diff
@@ -307,8 +307,8 @@ struct DecodeSlicesTask : BaseTask {
     std::shared_ptr<std::unordered_set<std::string>> filter_columns_;
 
     DecodeSlicesTask(
-            const StreamDescriptor& desc,
-            const std::shared_ptr<std::unordered_set<std::string>>& filter_columns)  :
+            const StreamDescriptor desc,
```

Comment:
I disagree, StreamDescriptor is just a wrapper of a shared pointer containing the photo data. Copy is fine.

    std::shared_ptr<Proto> data_ = std::make_shared<Proto>();

Reply:
I'm not fully aware of the use cases and the whole situation, but in general if we can avoid copying a shared_ptr we should do it (especially in performance critical code). They use atomics for ref counting which makes copying shared_ptr ~100 times slower than a bare ptr. Also the memory segment where the ref count is placed is in a dynamically allocated memory which requires one pointer jump to reach it. So if this part of the "hot path" and it's being called a lot of times I'd think of another option. If this is a rarely called function and not perf critical it's fine(ish). My personal opinion about shared ptr is to avoid using them, unless absolutely necessary.